netgrasp/utils/mac.py

```diff
@@ -0,0 +1,255 @@
+        db = database.Database(dbfile, lockfile, debugger)
+        source = db.get_state(DB_LOOKUP_CONFIG_SOURCE_KEY)
+        updated = db.get_state(DB_LOOKUP_CONFIG_UPDATED_KEY, is_date=True)
+        if source is None or updated is None:
+            raise Exception('config values not found')
+        config = {}
+        config['source'] = source
+        config['updated'] = updated
+    except Exception as e:
+        return False, 'error reading mac lookup config: %s' % e
+
+    return True, config
+
+
+def write_lookup_config(dbfile, lockfile, debugger, source=None, updated=None):
+    from netgrasp.database import database
+
+    source = source or DEFAULT_LOOKUP_SOURCE
+    updated = updated or DEFAULT_LOOKUP_UPDATED
+
+    if source not in LOOKUP_SOURCES:
+        return False, 'invalid mac lookup source "%s", must be one of: %r' % (source, LOOKUP_SOURCES)
+
+    try:
+        db = database.Database(dbfile, lockfile, debugger)
+        db.set_state(DB_LOOKUP_CONFIG_SOURCE_KEY, source)
+        db.set_state(DB_LOOKUP_CONFIG_UPDATED_KEY, updated)
+    except Exception as e:
+        return False, 'error writing mac lookup config: %s' % e
+
+    return True, dict(source=source, updated=updated)
+
+
+def get_dbupdated_text(updated):
+    text = 'never' if updated == DEFAULT_LOOKUP_UPDATED else '%s' % updated
+    return text
+
+
+def import_data(dbdata, dbfile, lockfile, debugger):
+    from netgrasp.database import database
+
+    debugger.info('importing data')
+    db = database.Database(dbfile, lockfile, debugger)
+    db.cursor.execute("DELETE FROM %s" % DB_LOOKUP_TABLE_NAME)
+    for values in dbdata:
+        values[0] = normalize_mac(values[0])
+        db.cursor.execute("INSERT INTO %s (address, netmask, name, fullname) VALUES (?, ?, ?, ?)" %
+                          DB_LOOKUP_TABLE_NAME, values)
+    db.connection.commit()
+    debugger.info('import successful')
+    db.set_state(DB_LOOKUP_CONFIG_UPDATED_KEY, datetime.datetime.now())
+
+
+def parse_wireshark_data(data):
+    """Return list of parsed manuf lines."""
+    dbdata = []
+    for line in data.splitlines():
+        values = parse_wireshark_line(line)
+        if values is None:
+            continue
+        dbdata.append(values)
+    return dbdata
+
+
+def parse_wireshark_line(line):
+    """Return list of parsed fields or None if empty line, comment or error."""
+    line = line.strip()
+    if not line or line[0] == '#':
+        return None
+    fields = line.split()
+    # fields: ['00:1B:C5:00:00:00/36', 'Convergi', '#', 'Converging', 'Systems', 'Inc.']
+    if len(fields) < 2:
+        return None
+    address = fields[0]
+    addrmask = address.split('/')
+    if len(addrmask) == 2:
+        address = addrmask[0]
+        netmask = int(addrmask[1])
+    else:
+        netmask = 0
+    name = fields[1]
+    fullname = ' '.join(fields[3:]) if len(fields) >= 4 else None
+    return [address, netmask, name, fullname]
+
+
+def download_wireshark_data(debugger):
```

Comment:
how long does this take? as noted before, I think all of this is a good fit for living in its own thread